src/lib.rs

```diff
@@ -505,12 +633,137 @@ Resources:
             );
+            let domain_binding = fields
+                .as_object()
+                .unwrap()
+                .get("domainBinding")
+                .map_or(None, |domain_binding| {
+                    Authority::from_str(domain_binding.as_str().unwrap()).ok()
+                });
+            let match_nonce = fields
+                .as_object()
+                .unwrap()
+                .get("matchNonce")
+                .map_or(None, |match_nonce| match_nonce.as_str());
+            let timestamp = fields
+                .as_object()
+                .unwrap()
+                .get("time")
+                .map_or(None, |timestamp| {
+                    DateTime::<Utc>::from_str(timestamp.as_str().unwrap()).ok()
+                });
             assert!(
                 message.is_err()
                     || signature.is_err()
-                    || message.unwrap().verify(signature.unwrap()).is_err()
+                    || message
+                        .unwrap()
+                        .verify(
+                            signature.unwrap(),
+                            domain_binding.as_ref(),
+                            match_nonce,
+                            timestamp.as_ref(),
+                        )
+                        .is_err()
             );
             println!("✅")
         }
     }
+
+    const VALID_CASES: &'static [&'static str] = &[
+        // From the spec:
+        // All caps
+        "0x52908400098527886E0F7030069857D2E4169EE7",
+        "0x8617E340B3D01FA5F11F306F4090FD50E238070D",
+        // All Lower
+        "0xde709f2102306220921060314715629080e2fb77",
+        "0x27b1fdb04752bbc536007a920d24acb045561c26",
+        "0x5aAeb6053F3E94C9b9A09f33669435E7Ef1BeAed",
+        "0xfB6916095ca1df60bB79Ce92cE3Ea74c37c5d359",
+        "0xdbF03B407c01E7cD3CBea99509d93f8DDDC8C6FB",
+        "0xD1220A0cf47c7B9Be7A2E6BA89F429762e7b9aDb",
+    ];
+
+    const INVALID_CASES: &'static [&'static str] = &[
+        // From eip55 Crate:
+        "0xD1220a0cf47c7B9Be7A2E6BA89F429762e7b9aDb",
+        "0xdbF03B407c01e7cD3CBea99509d93f8DDDC8C6FB",
+        "0xfb6916095ca1df60bB79Ce92cE3Ea74c37c5D359",
+        "0x5aAeb6053f3E94C9b9A09f33669435E7Ef1BeAed",
+        // FROM SO QUESTION:
+        "0xCF5609B003B2776699EEA1233F7C82D5695CC9AA",
+        // From eip55 Crate Issue
+        "0x000000000000000000000000000000000000dEAD",
+    ];
+
+    #[test]
+    fn test_is_checksum() {
+        for case in VALID_CASES {
+            let c = case.trim_start_matches("0x");
+            assert_eq!(is_checksum(&c), true)
+        }
+
+        for case in INVALID_CASES {
+            let c = case.trim_start_matches("0x");
+            assert_eq!(is_checksum(&c), false)
+        }
+    }
+
+    #[test]
+    fn eip55_test() {
+        // vectors from https://github.com/ethereum/EIPs/blob/master/EIPS/eip-55.md
+
+        assert!(test_eip55(
+            "0x5aAeb6053F3E94C9b9A09f33669435E7Ef1BeAed",
```

Comment:
Should these test cases be changed so that they don't already match the expectation? For example call `to_lowercase` in `test_eip55`?

Reply:
I changed `test_eip55` to produce and compare the checksum against the original string, the lowercased version and the uppercased version.